x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealm.java

```diff
@@ -57,21 +58,25 @@
     public static final String NAME = "reserved";
 
     private final ReservedUserInfo bootstrapUserInfo;
+    private final ReservedUserInfo autoconfigUserInfo;
     public static final Setting<SecureString> BOOTSTRAP_ELASTIC_PASSWORD = SecureSetting.secureString("bootstrap.password",
             KeyStoreWrapper.SEED_SETTING);
-    public static final Setting<SecureString> AUTOCONFIG_BOOOTSTRAP_ELASTIC_PASSWORD_HASH =
-        SecureSetting.secureString("autoconfig.password_hash", null);
+
+    // we do not document this setting on the website because it mustn't be set by the users
+    // it is only set by various installation scripts
+    public static final Setting<SecureString> AUTOCONFIG_ELASTIC_PASSWORD_HASH =
+        SecureSetting.secureString("autoconfiguration.password_hash", null);
 
     private final NativeUsersStore nativeUsersStore;
     private final AnonymousUser anonymousUser;
     private final boolean realmEnabled;
     private final boolean anonymousEnabled;
-    private final SecurityIndexManager securityIndex;
+    private final boolean elasticUserAutoconfigured;
 
     private final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(logger.getName());
 
     public ReservedRealm(Environment env, Settings settings, NativeUsersStore nativeUsersStore, AnonymousUser anonymousUser,
-                         SecurityIndexManager securityIndex, ThreadPool threadPool) {
+                         ThreadPool threadPool) {
```

Comment:
Accounting for if the security index exists is redundant because the native users store does it already.